drivers/usb_c/tcpc/shell.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/usb_c/usbc.h>
+#include <zephyr/shell/shell.h>
+
+/** Macro used to call the dump_std_reg function from USB-C connector node */
+#define TCPC_DUMP_NODE(node) ret |= tcpc_dump_std_reg(DEVICE_DT_GET(DT_PROP(node, tcpc)));
+
+/**
+ * @brief Shell command that dumps standard registers of TCPCs for all available USB-C ports
+ *
+ * @param sh Shell structure
+ * @param argc Unused arguments count
+ * @param argv Unused arguments
+ * @return int ORed return values of all the functions executed, 0 in case of success
+ */
+static int cmd_tcpc_dump(const struct shell *sh, size_t argc, char **argv)
+{
+	int ret = 0;
+
+	DT_FOREACH_STATUS_OKAY(usb_c_connector, TCPC_DUMP_NODE);
```

Comment:
Instead of always displaying every port, I suggest adding a parameter that specifies the port using the device handle.

You could use (SHELL_DYNAMIC_CMD_CREATE)[https://github.com/zephyrproject-rtos/zephyr/blob/14348c53d41c5f080648b71f8fa1817b0da6b304/drivers/i2c/i2c_shell.c#L320C1-L320C25] with a lookup function to provide tab completion for valid device handles.

Reply:
I did version with and without parameters, without parameters prints all devices. Is it ok (i2c did the same) that the completion suggests all devices instead of the only valid ones?

Reply:
It's okay for tab completion to show all devices, as there aren't any attributes that specifically identify the TCPC/VBUS devices.

I'm okay with "tcpc dump" showing all devices and "tcpc dump <device>" displaying a single device.  I don't see any need to support "tcpc dump <device1> <device2> ..."

Reply:
Right, no need to use this command with multiple devices, removed the foreach.